store/tikv/2pc.go

```diff
@@ -850,15 +888,78 @@ type RelatedSchemaChange struct {
 	Amendable        bool
 }
 
-func (c *twoPhaseCommitter) checkSchemaValid() error {
-	checker, ok := c.txn.us.GetOption(kv.SchemaChecker).(schemaLeaseChecker)
-	if ok {
-		_, err := checker.CheckBySchemaVer(c.commitTS, c.txn.txnInfoSchema)
+func (c *twoPhaseCommitter) tryAmendTxn(ctx context.Context, startInfoSchema SchemaVer, change *RelatedSchemaChange) (bool, error) {
+	memAmended := false
+	addMutations, err := c.txn.schemaAmender.AmendTxn(ctx, startInfoSchema, change, c.mutations)
+	if err != nil {
+		return false, err
+	}
+	// Prewrite new mutations.
+	if addMutations != nil && len(addMutations.keys) > 0 {
+		prewriteBo := NewBackofferWithVars(ctx, PrewriteMaxBackoff, c.txn.vars)
+		err = c.prewriteMutations(prewriteBo, *addMutations)
 		if err != nil {
-			return errors.Trace(err)
+			logutil.Logger(ctx).Warn("amend prewrite has failed", zap.Error(err), zap.Uint64("txnStartTS", c.startTS))
+			return false, err
 		}
+		logutil.Logger(ctx).Info("amend prewrite finished", zap.Uint64("txnStartTS", c.startTS))
+		memAmended = true
```

Comment:
Could we return true directly? Other places all return directly, and we can remove `memAmended`.